### PR TITLE
Load depth data from tmp(ts) sprites.

### LINKED ISSUE #5350: Tiberian Sun / RA2 tileset support [$130]
This is a tracking bug for implementing the isometric heightmapped terrain used in TS and RA2.  This is the main prerequisite of our TS mod (#3363).

Requirements (roughly in the order they will need to be completed):

- [x] Introduce a new self-consistent 3d coordinate system. [#2887, #2917, #3433, #3434, #3531, #3583, #3586, #3592, #3861, #3886, #3887, #4371, plus probably others]
- [x] Unhardcode the infantry sub-cells to support 3 instead of 5 and adapt the perspective [#6100]
- [x] Add support for the tmp(TS) terrain tiles.
  - [x] Update the terrain renderer to support additional sprite types. [#4185]
  - [x] Add support for flat isometric tiles. [#4391]
  - [x] Add support for loading embedded overlay artwork (cliffs, slopes). [#6855]
  - [x] Add support for TS/RA2 style terrain variants. [#7757]
  - [x] Support parsing the depth data as additional sprite frames. [[pchote:ts-tileset-depth](https://github.com/pchote/OpenRA/tree/ts-tileset-depth)]
- [x] Add support for orthographic (isometric) tile renderering.
  - [x] Support rectangular tiles with perspective. [#4391]
  - [x] Generalize the (currently hardcoded) relationship between cell and world coordinates.
[#5629, #5757, #5758]
  - [x] Introduce isometric coordinate calculations. [#5778]
- [x] Add support for heightmaps and slope tiles.
  - [x] Extend Tileset format with fields for slope type and LAT relationships. [#6855]
  - [x] Extend Map format with a heightmap block. [#6901]
  - [x] Update cell -> world coordinate conversion to use the heightmap data. [#6901]
- [x] Write a TS/RA2 tileset.ini converter [#6708].
- [x] Write a new map editor that can render and place isometric tiles. [#4496]

<bountysource-plugin>
---
There is a **[$130 open bounty](https://www.bountysource.com/issues/2211547-tiberian-sun-ra2-tileset-support-130?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pchote**
This exposes the depth data as additional frames so that we can handle them similarly to split shadows. https://github.com/pchote/OpenRA/commits/depthbuffer shows how this will eventually be used (the way it loads the data is stable, but the rest is still very experimental).

This also fixes an issue where the asset browser doesn't list half of the available palettes.

Closes #5350.

### CODE REVIEW

**Comment by RoosterDragon** [comment]
Question: What's the benefit to making PaletteNames a sequence, as opposed to a single name?

**Comment by pchote** [comment]
`ILoadsPalettes` in principle allows one trait to load multiple palettes, and it simplifies the case where a trait may not return any valid palettes (see PaletteFromFile).

File: OpenRA.Mods.TS/SpriteLoaders/TmpTSLoader.cs [hunk 1]
```diff
@@ -67,8 +85,16 @@ public TmpTSFrame(Stream s, Size size, int u, int v)
 						width += (j < size.Height / 2 - 1 ? 1 : -1) * 4;
 					}
 
-					// TODO: Load Z-data once the renderer can handle it
-					s.Position += size.Width * size.Height / 2;
+					// Unpack depth data
```
Comment (penev92) [inline OpenRA.Mods.TS/SpriteLoaders/TmpTSLoader.cs]: The following code is identical to the unpacking of the tile data just above. How about extracting it into a method?

**Comment by Mailaender** [comment]
I can confirm that I see more palettes and tileset frames in the TS asset browser. :white_check_mark:

**Comment by reaperrr** [comment]
> OpenRA.Mods.TS/SpriteLoaders/TmpTSLoader.cs:L79: [SP2000] Invalid spacing at the end of the line.

**Comment by pchote** [comment]
Fixed. #8015 makes these errors a pain for me to check locally.

**Comment by obrakmann** [comment]
:+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/5a10ff1763b03d111b3d2d990adb9f59c6989942%5E...5a10ff1763b03d111b3d2d990adb9f59c6989942)

**Pull request merged**